File: src/main/resources/static/js/property.js

$(function () {
    $("#submit").on("click", function () {
        var name = $("#name").val();
        var address = $("#address").val();
        var processingTime = $("#processingTime").val();
        var number = $("#number").val();
        var cases = $("#cases").val();
        var stand = $("#stand").val();
        var unitNumber= $("#unitNumber").val();
        var typeOfRight = $("#typeOfRight").val();
        var natureOfRight = $("#natureOfRight").val();
        //获取用途的值
        var purpose = $("#purpose").val();
       //将字符串转换为毫秒数
        var purposeTime = purpose*365*24*60*60*1000;
        var usableArea = $("#usableArea").val();
        var coveredArea = $("#coveredArea").val();

        var structure = $("#structure").val();
        var totalFloors =$("#totalFloors").val();
        var storyNumber = $("#storyNumber").val();
        var privateUseArea = $("#privateUseArea").val();
        var exclusiveAcreage = $("#exclusiveAcreage").val();
        var mjft= $("#mjft").val();
        var exclusiveFloorSpace = $("#exclusiveFloorSpace").val();
        var apportionedFloorSpace = $("#apportionedFloorSpace").val();

        //获取登记的时间的值
        var registrationTime = $("#registrationTime").val();
        var registrationTimes = (new Date(registrationTime)).getTime(); //得到毫秒数


        var lifeSpanTime = purposeTime+registrationTimes;//使用时期毫秒数
        var lifeSpan = new Date(lifeSpanTime).toLocaleDateString();//转换时间格式
        var abbreviation = $("#abbreviation").val();



        $.ajax({
            type: "POST",
            url: "/property",
            data: { name,address,processingTime,number,cases,stand,unitNumber,typeOfRight,natureOfRight,purpose,
                usableArea,coveredArea,structure,totalFloors,storyNumber,privateUseArea,exclusiveAcreage,mjft,
                exclusiveFloorSpace,apportionedFloorSpace,registrationTime,lifeSpan,abbreviation},
            statusCode: {
                201: function () {
                    alert("信息提交成功");
                    $(location).attr("href", "/property.html");
                    return false;
                },
                500: function () {
                    alert("信息提交失败，请重新提交");
                    $(location).attr("href", "/property.html");
                    return false;
                }
            }
        });
    });
});

File: src/main/java/com/example/springboot_houseinfo/pojo/Property.java
package com.example.springboot_houseinfo.pojo;
import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;
import org.springframework.format.annotation.DateTimeFormat;

import java.time.LocalDate;
import java.util.Date;

@Data
@ToString
@AllArgsConstructor
@NoArgsConstructor
public class Property {
    private Integer id;
    private String name;
    private String address;
    @DateTimeFormat(pattern = "yyyy-MM-dd")
    @JsonFormat(pattern = "yyyy-MM-dd")
   /* 1、LocalDate是日期对象，每次操作都会回到新的时间对象。
   2、可以直接分析日期字符串，生成LocalDate实例，和LocalTime操作一样简单。*/
    private LocalDate processingTime;
    private String number;
    private String cases;
    private String  stand;
    private String unitNumber;
    private String typeOfRight;
    private String natureOfRight;
    private Integer purpose;
    private String usableArea;
    private String coveredArea;
    private String structure;
    private Integer totalFloors;
    private Integer storyNumber;
    private String privateUseArea;
    private String exclusiveAcreage;
    private String mjft;
    private String exclusiveFloorSpace;
    private String apportionedFloorSpace;
    @DateTimeFormat(pattern = "yyyy-MM-dd")
    @JsonFormat(pattern = "yyyy-MM-dd")
    private LocalDate registrationTime;
    private String lifeSpan;
    private String abbreviation;

}
